hs_core/HydroRealtimeSignalProcessor.py

```diff
@@ -0,0 +1,110 @@
+from django.db import models
+from haystack.signals import RealtimeSignalProcessor
+from haystack.exceptions import NotHandled
+import logging
+import types
+
+logger = logging.getLogger(__name__)
+logger.debug("load real time processor")
+
+
+class HydroRealtimeSignalProcessor(RealtimeSignalProcessor):
+
+    """ 
+    Customized for the fact that all indexed resources are subclasses of BaseResource. 
+
+    Notes: 
+    1. RealtimeSignalProcessor already plumbs in all class updates. We might want to be more specific. 
+    2. The class sent to this is a subclass of BaseResource, or another class. 
+    3. Thus, we want to capture cases in which it is an appropriate instance, and respond. 
+    """
+
+    def hydro_resource_is_present(self, instance): 
+        """ 
+        This returns True if a current resource is now present in Haystack, and False if not. 
+        This -- in turn -- informs us as to whether to update and/or delete the object 
+        """
+        from haystack.query import SearchQuerySet
+        from haystack.utils import get_identifier
```

Comment:
Yes. I agree there is no circular import issue for this case. But we should do imports inside methods "handle_save" and "handle_delete", right?